sdks/java/io/hadoop-input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/unit/tests/inputs/NewObjectsEmployeeInputFormat.java

```diff
@@ -0,0 +1,188 @@
+			inputSplitList.add(inputSplitObj);
+		}
+		return inputSplitList;
+	}
+
+	public static class NewObjectsEmployeeInputSplit extends InputSplit implements
+			Writable {
+		// Start and end map index of each split of employeeData.
+		private long startIndex;
+		private long endIndex;
+
+		public NewObjectsEmployeeInputSplit() {
+		}
+
+		public NewObjectsEmployeeInputSplit(long startIndex, long endIndex) {
+			this.startIndex = startIndex;
+			this.endIndex = endIndex;
+		}
+
+		/**
+		 * Returns number of records in each split.
+		 */
+		@Override
+		public long getLength() throws IOException, InterruptedException {
+			return this.endIndex - this.startIndex + 1;
+		}
+
+		@Override
+		public String[] getLocations() throws IOException, InterruptedException {
+			return null;
+		}
+
+		public long getStartIndex() {
+			return startIndex;
+		}
+
+		public long getEndIndex() {
+			return endIndex;
+		}
+
+		@Override
+		public void readFields(DataInput dataIn) throws IOException {
+			startIndex = dataIn.readLong();
+			endIndex = dataIn.readLong();
+		}
+
+		@Override
+		public void write(DataOutput dataOut) throws IOException {
+			dataOut.writeLong(startIndex);
+			dataOut.writeLong(endIndex);
+		}
+	}
+
+	public class NewObjectsEmployeeRecordReader extends
+			RecordReader<Text, Employee> {
+
+		private NewObjectsEmployeeInputSplit split;
+		private Text currentKey;
+		private Employee currentValue;
+		private long employeeListIndex = 0L;
+		private long recordsRead = 0L;
+		private List<KV<String, String>> employeeDataList;
+
+		public NewObjectsEmployeeRecordReader() {
+		}
+
+		@Override
+		public void close() throws IOException {
+		}
+
+		@Override
+		public Text getCurrentKey() throws IOException, InterruptedException {
+			return currentKey;
+		}
+
+		@Override
+		public Employee getCurrentValue() throws IOException,
+				InterruptedException {
+			return currentValue;
+		}
+
+		@Override
+		public float getProgress() throws IOException, InterruptedException {
+			return (float) recordsRead / split.getLength();
+		}
+
+		@Override
+		public void initialize(InputSplit split, TaskAttemptContext arg1)
+				throws IOException, InterruptedException {
+			this.split = (NewObjectsEmployeeInputSplit) split;
+			employeeListIndex = this.split.getStartIndex() - 1;
+			recordsRead = 0;
+			employeeDataList = TestEmployeeDataSet.populateEmployeeData();
+			currentValue = new Employee(null, null);
+		}
+
+		@Override
+		public boolean nextKeyValue() throws IOException, InterruptedException {
+			if ((recordsRead++) >= split.getLength()) {
+				return false;
+			}
+			employeeListIndex++;
+			KV<String, String> employeeDetails = employeeDataList
+					.get((int) employeeListIndex);
+			String empData[] = employeeDetails.getValue().split("_");
+			/*
+			 * New objects returned every time for key and value to signify the same
+			 * object's state is not changed and returned instead every time a new object is returned.
```

Comment:
replace this comment with:
New objects must be returned every time for key and value in order to test the scenario as discussed the in the class' javadoc.